software_training_assignment/src/goto_action.h

```diff
@@ -0,0 +1,97 @@
+#include <software_training_assignment/GotoAction.h>
+#include <actionlib/server/simple_action_server.h>
+#include <turtlesim/Pose.h>
+#include <geometry_msgs/Twist.h>
+
+namespace A
+{
+    class GotoAction
+    {
+    	public:
+    		ros::NodeHandle node;
+    		GotoAction(std::string name) : as_(node, name, boost::bind(&GotoAction::executeCB, this, _1), false)
+	    	{
+	    		as_.start();
+				ROS_INFO("Action server started");
+	    	}
+
+	    	~GotoAction(void)
+	    	{
+	    	}
+
+	    	ros::Publisher pub;
+	    	ros::Subscriber sub;
+	    	double x, y;
+    
+	    	void subCallback(const turtlesim::Pose::ConstPtr& msg)
+	    	{
+		    	x = msg->x;
+		    	y = msg->y;
+	    	}
+
+	    	void executeCB(const software_training_assignment::GotoGoal::ConstPtr &goal)
+	    	{
+			
+				ROS_INFO("executeCB called");
```

Comment:
this formatting is all off, make sure to format properly so that it's easy to read the code